presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/RewriteAggregationIfToFilter.java

```diff
@@ -199,4 +225,23 @@ private boolean shouldRewriteAggregation(Aggregation aggregation, ProjectNode so
         // Only rewrite the aggregation if the else branch is not present or the else result is NULL.
         return expression.getForm() == IF && Expressions.isNull(expression.getArguments().get(2));
     }
+
+    private boolean canUnwrapIf(SpecialFormExpression ifExpression)
+    {
+        boolean[] result = {true};
+        ifExpression.getArguments().get(1).accept(new DefaultRowExpressionTraversalVisitor<boolean[]>()
```

Comment:
To be safe, let's also disable it for `visitLambda`. I don't exactly what will happen to lambda. It's a corner case; so maybe just disable it with a comment.

Reply:
Done.